### PR TITLE
Add analysis data archive bucket

### PR DESCRIPTION
**Pull request by reisingerf**
This will add an analysis data archive bucket in our UniMelb data account.

Note: 
- There is no write access configured (yet) and other access only for the FileManager ingest role.
- Versioning is enabled (since production data)
- All objects get immediately pushed to GLACIER_IR storage class

### CODE REVIEW

File: terraform/stacks/unimelb/data_archive/analysis_archive.tf [hunk 1]
```diff
@@ -0,0 +1,101 @@
+################################################################################
+# Local constants
+
+locals {
+  # The bucket holding all archived analysis data
+  analysis_archive_bucket_name = "archive-prod-analysis-${data.aws_caller_identity.current.account_id}-${data.aws_region.current.name}"
```
Comment (mmalenic) [inline terraform/stacks/unimelb/data_archive/analysis_archive.tf:6]: This bucket will need to be configured as an event source in orcabus too, if filemanager wants to ingest events?

**Review by victorskl — APPROVED** [review]
LGTM

File: terraform/stacks/unimelb/data_archive/analysis_archive.tf (hunk 1 above)
Comment (reisingerf) [reply terraform/stacks/unimelb/data_archive/analysis_archive.tf:6]: Yes I forgot the event bus config... adding...

Comment (reisingerf) [reply terraform/stacks/unimelb/data_archive/analysis_archive.tf:6]: Added with https://github.com/umccr/infrastructure/pull/490/commits/d2ec42bb2ae8073eb2c0ba0da3a18ebf77a1d3b4

Note: this does not include the activation of the eventbridge integration at the bucket level (due to possible integration with ICA BYOB). Any subscriber (like the FileManager) will have to set up their own rules against the OrcaBus event bus.

Comment (mmalenic) [reply terraform/stacks/unimelb/data_archive/analysis_archive.tf:6]: Is the intent to have this contain just prod data? E.g. there won't be an `archive-dev-analysis-*` bucket?

Comment (reisingerf) [reply terraform/stacks/unimelb/data_archive/analysis_archive.tf:6]: Yes, we don't have plans to archive any `dev` data. That's a `prod` thing only.

**Comment by victorskl** [comment]
Gentle ping @alexiswl - pls review

**Review by alexiswl — APPROVED** [review]

**Pull request merged**